app/script/storage/StorageService.js

```diff
@@ -72,42 +78,25 @@ class StorageService {
     });
   }
 
-  _upgradeStores() {
+  _upgradeStores(db) {
     z.storage.StorageSchemata.SCHEMATA.forEach(({schema, upgrade, version}) => {
+      const versionUpdate = db.version(version).stores(schema);
       if (upgrade) {
-        return this.db
-          .version(version)
-          .stores(schema)
-          .upgrade(transaction => {
-            this.logger.warn(`Database upgrade to version '${version}'`);
-            upgrade(transaction, this.db);
-          });
+        versionUpdate.upgrade(transaction => {
+          this.logger.warn(`Database upgrade to version '${version}'`);
+          upgrade(transaction, db);
+        });
       }
-
-      this.db.version(version).stores(schema);
     });
   }
 
   // Hooks
   addUpdatedListener(storeName, callback) {
-    this.updateListeners.push(callback);
-    const initHook = () => {
-      if (this.updateListeners.length > 1) {
-        const updateListeners = this.updateListeners;
-        this.db[storeName].hook('updating', function(modifications, primaryKey, previousRecord, transaction) {
-          // we need to wait for the transaction to be finished in order to be able to access the DB later on
-          this.onsuccess = updatedRecord =>
-            transaction.on('complete', () =>
-              updateListeners.forEach(callbackFn => callbackFn(updatedRecord, previousRecord))
-            );
-        });
-      }
-    };
-    if (!this.db) {
-      // waiting for the DB to be initialized
-      return (this._afterDbInit = initHook);
-    }
-    initHook();
+    this.dbListeners.push({callback, store: storeName, type: 2});
```

Comment:
What's `type: 2`?

Reply:
This is the type of change we want to listen to. These are value sent by Dexie.
Do you think I should create a named variable to make that more explicit?

Reply:
That would be great.